test/logger_test.rb

```diff
@@ -0,0 +1,34 @@
+# typed: true
+# frozen_string_literal: true
+
+require_relative "./test_helper"
+
+module ShopifyAPITest
+  class LoggerTest < Minitest::Test
+    def test_call_warn
```

Comment:
Is this a useful test? It kind of just tests if `send_to_logger` was called the correct message and log_level. I didn't make more of these because I wanted to see what we thought of this. (#still-a-noob)

Reply:
Good question! IMHO `send_to_logger` should be a private method so this would break my rule of expecting messages from ourselves. I'd put the expectation on the context logger:

```ruby
ShopifyAPI::Context.logger.expects(:warn, "test")
```

The behavior we are looking at testing is that our logger should be warning with the string of "test" so that is what the test is here.

If you ended up leaving `send_to_logger` as a public method, then this test would be fine :) I'd argue it is a private function because you wouldn't call that method outside of the context of that class because it would require users of this class to know exactly how logging is wired together.

Reply:
Happy to pair on this if you'd like :) Testing is an abstract art that takes some experience shooting yourself in the foot to really get a grasp on